internal/common/validation/podspec.go

```diff
@@ -54,6 +59,35 @@ func ValidatePodSpec(spec *v1.PodSpec, schedulingSpec *configuration.SchedulingC
 	return validatePorts(spec)
 }
 
+func validateTerminationGracePeriod(spec *v1.PodSpec, config *configuration.SchedulingConfig) error {
+	hasTerminationGracePeriod := spec.TerminationGracePeriodSeconds != nil
+	var terminationGracePeriodSeconds int64
+	var exceedsBounds, isDefault bool
+	if hasTerminationGracePeriod {
+		terminationGracePeriodSeconds = *spec.TerminationGracePeriodSeconds
+		exceedsBounds = (terminationGracePeriodSeconds < int64(config.TerminationGracePeriod.Minimum.Seconds()) ||
+			terminationGracePeriodSeconds > int64(config.TerminationGracePeriod.Maximum.Seconds()))
+		isDefault = terminationGracePeriodSeconds == int64(config.TerminationGracePeriod.Default)
+	}
+
+	if config.Preemption.Enabled &&
+		hasTerminationGracePeriod &&
+		exceedsBounds {
+
+		if isDefault {
```

Comment:
I know you got asked to do this- but imo this is nuts!  We shouldn't validate this here but rather just validate the config on startup and bail if something crazy has been set.  I'm pretty sure we don't validate any config on startup right now so we probably need a separate issue to get this done.